### PR TITLE
clock_gettime, nanosleep and gettimeofday syscalls

### PR DESCRIPTION
**Pull request by czapiga**
Implementing syscalls from #363

### CODE REVIEW

**Comment by cahirwpz** [comment]
@czapiga What is the progress on the task? Have you read my recent comment in #363?

**Comment by czapiga** [comment]
@cahirwpz I will work on that in next three days (starting tomorrow). Sorry but I am very busy with my work - I had many overtime hours this week including today. I think I will end it before our next meeting.

**Comment by czapiga** [comment]
@cahirwpz Well, I'm not sure if I'm doing it right. I'm calling `do_clock_gettime` in `user/libmimiker/time.c` instead of `clock_gettime` because there is no `clock_gettime` function for now. I need `time.h` header and I'm not sure if I should just link our `include/time.h` to `user/newlib/overrides/include/` and make `#ifdef _KERNELSPACE` or what..

File: sys/sysent.c [hunk 1]
```diff
@@ -362,6 +363,21 @@ static int sys_access(thread_t *td, syscall_args_t *args) {
   return result;
 }
 
+static int sys_clock_gettime(thread_t *td, syscall_args_t *args) {
+  clockid_t clk = (clockid_t)args->args[0];
+  timespec_t *ts = (timespec_t *)args->args[1];
```
Comment (cahirwpz) [inline sys/sysent.c]: It is **strictly prohibited** to operate on userspace memory from within kernel code. You **must** use `copyin` and `copyout` routines to safely move data between kernelspace and userspace.

File: sys/time.c [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#include <errno.h>
+#include <thread.h>
+#include <time.h>
+#include <timer.h>
+#include <sync.h>
+
+int do_clock_gettime(clockid_t clk, timespec_t *tp) {
+  if (tp == NULL)
+    return -EFAULT;
```
Comment (cahirwpz) [inline sys/time.c]: I fixed your embarrasing failure to recognize how we return `errno` values to userspace.

File: user/libmimiker/time.c [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#include <time.h>
+
+int nanosleep(timespec_t *rqtp, timespec_t *rmtp) {
+  return do_clock_nanosleep(CLOCK_REALTIME, 0, rqtp, rmtp);
+}
+
+int do_gettimeofday(timeval_t *tp, void *tzp) {
```
Comment (cahirwpz) [inline user/libmimiker/time.c]: Why `do_gettimeofday` and not `gettimeofday`?

**Comment by czapiga** [comment]
@cahirwpz I waited with tests to the point when implementation would be ready. I just wanted to meet with you and discuss few things (related to syscalls implementation). Also please check my mails I sent you in last 2 days.

**Comment by cahirwpz** [comment]
@melouber & @ilikeheaps After you're done with callouts and interruptible sleep I'd like you to finish this PR. Please note the change adds user-space visible functionality to the kernel!

**Pull request merged**